src/main/java/io/dataspaceconnector/services/usagecontrol/RuleValidator.java

```diff
@@ -266,10 +283,84 @@ public void validatePurpose(final Rule rule, String consumerURI)
         } catch (UnsupportedEncodingException e) {
         }
 
-       if (!allowedRoleAsString.equals(consumerPurposeAsString)) {
+       if (!allowedPurposeAsString.equals(consumerPurposeAsString)) {
             throw new PolicyRestrictionException(ErrorMessages.DATA_ACCESS_INVALID_PURPOSE);
        }
     }
-    
+
+
+
+    //TECNALIA-ICT-OPTIMA: New method.
+    /**
+     * Checks whether the requesting connector corresponds to the allowed connector.
+     *
+     * @param rule        The ids rule.
+     * @param consumerURI The consumer URI.
+     * @throws PolicyRestrictionException If it is not allowed to access the data with such Purpose.
+     */
+    public void validateIssuerConnector(final Rule rule, String consumerURI)
+            throws PolicyRestrictionException {
+
+        String allowedConsumer = RuleUtils.getEndpoint(rule);
+        //URI allowedConsumerURI = getConnectorRestriction(rule);
+        URI allowedConsumerURI=URI.create(allowedConsumer);
+        if (allowedConsumerURI == null) {
+            if (log.isWarnEnabled()) {
+                log.warn("Purpose is null. [rule=({})]", rule);
+            }
+            throw new PolicyRestrictionException(ErrorMessages.DATA_ACCESS_INVALID_CONSUMER);
+        }
+
+        String allowedConsumerURIAsString = allowedConsumerURI.toString();
+
+        if (!allowedConsumerURIAsString.equals(consumerURI)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Invalid consumer connector. [issuer=({})]", consumerURI);
+            }
+            throw new PolicyRestrictionException(ErrorMessages.DATA_ACCESS_INVALID_CONSUMER);
+        }
+
+      /*  String allowedRoleAsString = allowedConsumerURI.toString();
+        String consumerPurposeAsString = consumerURI.toString();
+
+        if (!allowedRoleAsString.equals(consumerPurposeAsString)) {
+            throw new PolicyRestrictionException(ErrorMessages.DATA_ACCESS_INVALID_CONSUMER);
+        }*/
+    }
+
+
+    /**
+     * Checks whether the requesting connector has the right security level.
+     *
+     * @param rule    The ids rule.
+     * @param profile The security profile.
+     * @throws PolicyRestrictionException If the connector ids do no match.
+     */
+
+    private void validateSecurityProfile(final Rule rule, final Optional<SecurityProfile> profile)
```

Comment:
Is this a proper way to validate the security profile?

Should that be fetched from the consumer connector self-description?

Reply:
I am investigating it to change the logic so if you any suggestion kindly share.